provider/manifest/watchdog.go

```diff
@@ -0,0 +1,72 @@
+package manifest
+
+import (
+	"context"
+	"github.com/boz/go-lifecycle"
+	"github.com/ovrclk/akash/provider/session"
+	dtypes "github.com/ovrclk/akash/x/deployment/types"
+	"github.com/ovrclk/akash/x/market/types"
+	"github.com/tendermint/tendermint/libs/log"
+	"time"
+)
+
+type watchdog struct {
+	leaseID types.LeaseID
+	timeout time.Duration
+	lc      lifecycle.Lifecycle
+	sess    session.Session
+	ctx     context.Context
+	log     log.Logger
+}
+
+func newWatchdog(sess session.Session, parent <-chan struct{}, done chan<- dtypes.DeploymentID, leaseID types.LeaseID, timeout time.Duration) *watchdog {
+	ctx, cancel := context.WithCancel(context.Background())
+	result := &watchdog{
+		leaseID: leaseID,
+		timeout: timeout,
+		lc:      lifecycle.New(),
+		sess:    sess,
+		ctx:     ctx,
+		log:     sess.Log().With("leaseID", leaseID),
+	}
+
+	go func() {
+		result.lc.WatchChannel(parent)
+		cancel()
+		result.lc.ShutdownAsync(nil)
+	}()
```

Comment:
this can just be 

```go
go result.lc.WatchChannel(parent)
```

Which will trigger a shutdown.

The newer context can be created in `run()`, but it looks like it might not really be necessary to create a new one.

Reply:
If I create a new context in `run()`, how can I cancel it when the parent channel closes?

Reply:
Do you need to create a new context?

In `run()`, you can do it in a defer and in the select I suppose.

Or do it like this, but the `ShutdownAsync()` is unnecessary - `WatchChannel` will do that.

Reply:
I'm passing it while doing the broadcast, presumably that should be cancelled when the parent closes. Otherwise it could block for a really long time.

Reply:
okay